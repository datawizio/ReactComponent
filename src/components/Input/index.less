@import "~antd/es/input/style/index";
@import "../../styles/variables";
