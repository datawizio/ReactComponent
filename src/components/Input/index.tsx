import React, { LegacyRef } from "react";

import { Input as AntInput } from "antd";
import { InputProps as AntInputProps, GroupProps } from "antd/lib/input";

import "./index.less";

export interface InputProps extends AntInputProps {}

export interface FCInput extends React.FC<InputProps> {
  Group: React.StatelessComponent<GroupProps>;
}

//@ts-ignore
const Input: FCInput = React.forwardRef((props, ref: LegacyRef<AntInput>) => {
  return <AntInput ref={ref} {...props} />;
});

Input.Group = AntInput.Group;

export default Input;
