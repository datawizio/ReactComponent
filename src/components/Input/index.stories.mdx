import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { UserOutlined } from "@ant-design/icons";
import Input from "./index";

<Meta title="Components/Input/Input" component={Input} />

# Input

Компонент `Input`.

Базовый компонент: [**antd/input**](https://ant.design/components/input/).

## Basic

<Preview>
  <Story name="Basic">
    <Input placeholder="Basic usage" />
  </Story>
</Preview>

## Large size

<Preview>
  <Story name="Large">
    <Input placeholder="Large size" size="large" />
  </Story>
</Preview>

## Small size

<Preview>
  <Story name="Small">
    <Input placeholder="Small size" size="small" />
  </Story>
</Preview>

## With prefix

<Preview>
  <Story name="With prefix">
    <Input placeholder="Prefix icon" prefix={<UserOutlined />} />
  </Story>
</Preview>

<Preview>
  <Input placeholder="Prefix text" prefix="Text" />
</Preview>

## With suffix

<Preview>
  <Story name="With suffix">
    <Input placeholder="Suffix icon" suffix={<UserOutlined />} />
  </Story>
</Preview>

<Preview>
  <Input placeholder="Suffix text" suffix="Text" />
</Preview>

## With clear icon

<Preview>
  <Story name="With clear icon">
    <Input placeholder="With clear icon" allowClear />
  </Story>
</Preview>

## Props

- [**antd/input**](https://ant.design/components/input/#API)

<Props of={Input} />
