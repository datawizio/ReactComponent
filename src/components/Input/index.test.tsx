import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import Input from "./index";

const mockProps = {};

const setUp = (props?) => mount(<Input {...props} />);

describe("Input component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("Input rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
