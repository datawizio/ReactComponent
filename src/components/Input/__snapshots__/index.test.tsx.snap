// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Input rendered correctly 1`] = `
<ForwardRef>
  <Input
    type="text"
  >
    <ClearableLabeledInput
      bordered={true}
      element={
        <input
          className="ant-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          type="text"
        />
      }
      focused={false}
      handleReset={[Function]}
      inputType="input"
      prefixCls="ant-input"
      triggerFocus={[Function]}
      type="text"
      value=""
    >
      <input
        className="ant-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value=""
      />
    </ClearableLabeledInput>
  </Input>
</ForwardRef>
`;
